### ISSUE TITLE
[Bug]: BuildCheck - building the solution with two projects that don't share the same output/intermediate output path has BC0101 warning

### ISSUE THREAD

**Issue by GangWang01** [dotnet/msbuild#10407]
[BCTest_Repro.zip](https://github.com/user-attachments/files/16281498/BCTest_Repro.zip) has the solution to reproduce the issue, as well as build binary log in the zip file. The solution has two simple library projects with default setting that locates build output at the project folder respectively. Build should have no BC0101 warning, but build result has the BC0101 warnings telling the projects share the same output path and intermediate output path which are not true.

From part of build log below, `C:\ws\test\BuildCheck\TestSln\BCTest_Repro` is the folder where the solution is. It looks like something wrong with the base output path of the projects during build check.
```
Build succeeded.

  MSBUILD : warning BC0101: https://aka.ms/buildcheck/codes#BC0101 - Projects ClassLibrary2.csproj and ClassLibrary1.cs
proj have conflicting output paths: C:\ws\test\BuildCheck\TestSln\BCTest_Repro\bin\Debug\net8.0\.
  MSBUILD : warning BC0101: https://aka.ms/buildcheck/codes#BC0101 - Projects ClassLibrary2.csproj and ClassLibrary1.cs
proj have conflicting output paths: C:\ws\test\BuildCheck\TestSln\BCTest_Repro\obj\Debug\net8.0\.

    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.84
```

**Comment by AR-May** [comment]
@GangWang01 this probably should be fixed by #10472 (I saw similar extra warnings in another repo and fixed it). 
Could you please verify that and close the issue if it is indeed true?

**Comment by GangWang01** [comment]
Yes, I verified with latest main. The issue was not reproed. It's fixed.
